test/pact_test.exs

```diff
@@ -68,4 +55,28 @@ defmodule PactTest do
     Pact.put(:string, Integer)
     assert Pact.get("string") == Integer
   end
+
+  test "replace module replaces the given module" do
+    Pact.put("foo", String)
+
+    Pact.replace self, "foo" do
+      def awesome? do
+        true
+      end
```

Comment:
Will this work if you define multiple functions? If so maybe we should add a test for that?